### PR TITLE
#6 comment course models

### PR DESCRIPTION
**Pull request by nicknapoli82**
Its a small one.

### CODE REVIEW

File: server/db/models/course.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+Course.init(
+  {
+    id: {
+      type: sequelize.UUID,
+      defaultValue: sequelize.UUIDV4,
+      unique: true,
+      primaryKey: true,
+    },
+    name: {
+      type: sequelize.STRING(1024),
```
Comment (blueteamcpu) [inline server/db/models/course.js:16]: Any particular reason for these string length choices?

Comment (nicknapoli82) [reply server/db/models/course.js:16]: Yes. The default string length is 255 characters, and I made my choices based on what I think may be the longest course name possible. Where 255 is quite possibly not long enough. 

I may be wrong. Its a bit of a safety measure though. Thats the only reason.

I should add. Often the way computers work, its best to define array, string, or similar array like structures in powers of two. Which in a way is why the default string length is actually 255, or a power of 2 - 1. There is significance in that design decision from sequelize and postgres. Maybe i hsould have stuck with the power of 2 -1 convention. But this is the reason.

Comment (nicknapoli82) [reply server/db/models/course.js:16]: Who knows. Maybe I'm overthinking a trivial thing. :man_shrugging:

Comment (justin-cook-developer) [reply server/db/models/course.js:16]: Sounds good to me! Thanks for explaining.

File: server/db/models/post.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    topic: {
+      type: sequelize.STRING,
+      unique: true,
+      allowNull: false,
+    },
+    description: {
+      type: sequelize.TEXT,
+      allowNull: true,
+    }
+  }
```
Comment (justin-cook-developer) [inline server/db/models/post.js:33]: All of these models need a second object like so, `{
    sequelize: db,
    modelName: 'user',
  }`, after the object which contains the column definitions. See my user model for a reference.

File: server/db/models/comment.js [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+  {
+    id: {
+      type: sequelize.UUID,
+      defaultValue: sequelize.UUIDV4,
+      unique: true,
+      primaryKey: true,
+    },
+    content: {
+      type: sequelize.TEXT,
+      allowNull: true,
```
Comment (justin-cook-developer) [inline server/db/models/comment.js]: I think we should require content. An empty comment is odd.

File: server/db/models/post.js [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+    id: {
+      type: sequelize.UUID,
+      defaultValue: sequelize.UUIDV4,
+      unique: true,
+      primaryKey: true,
+    },
+    topic: {
+      type: sequelize.STRING,
+      unique: true,
+      allowNull: false,
```
Comment (justin-cook-developer) [inline server/db/models/post.js:17]: I think we should add `validate: { notEmpty: true }` for topics and description

File: server/db/models/comment.js (hunk 3 above)
Comment (nicknapoli82) [reply server/db/models/comment.js]: Good point. I'll add that in this evening as a validation.

File: server/db/models/post.js (hunk 2 above)
Comment (nicknapoli82) [reply server/db/models/post.js:33]: Cool man. I missed that. I'll add that in as well this evening.
I'm used to using the Sequelize.define method. Didn't see that you need to add that sequelize db and model name in the class property for init.

**Review by justin-cook-developer — APPROVED** [review]

**Pull request merged**